### DISCUSSION TITLE
How do I prevent lots of same hcl files across many environments

### DISCUSSION THREAD

**Discussion by cveld** [gruntwork-io/knowledge-base#820] [How do I]
We have a folder structure:
```
├───_template
│   ├───_namespace
│   │   ├───component1
│   │   ├───component2
├───acc-weu-01
│   ├───acc
│   │   ├───component1
│   │   ├───component2
├───prd-weu-01
│   ├───prd
│   │   ├───component1
│   │   ├───component2
```

In all component folders there is this hcl file:
```
# for the component across all environments.
include "root" {
  path   = find_in_parent_folders()
  expose = true
}

# Include the common configuration for the component. The common configuration contains settings that are common
# for the component across all environments.
include "common" {
  path   = "${dirname(find_in_parent_folders())}/Infrastructure/${basename(get_terragrunt_dir())}/terragrunt.hcl"
  expose = true
}
```

Is it possible to run terragrunt against a list of virtual components? Ideally we should not have this hcl file copied many times across all components, all environment combinations.

---

<ins datetime="2024-04-25T11:29:15Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110903">Tracked in ticket #110903</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
if without a `terragrunt.hcl` file, Terragrunt will not be able to detect if this is a valid directory to use or not

**Comment by cveld** [comment]
@denis256 I think the product must be amended so that it will read `terragrunt.hcl` in the current directory or so to get directives?

**Reply by denis256** [reply]
Hm,
without `terragrunt.hcl`, it will not be clear in which directory to run Terragrunt or not

I think some workarounds can be implemented through `--terragrunt-strict-include / --terragrunt-include-dir`

**Comment by yhakbar** [comment]
Hey @cveld ,

Please take a look at this RFC and share your feedback there:
https://github.com/gruntwork-io/terragrunt/issues/3313

We're hoping this will address the kind of issue that you're facing here. I'm going to close out this discussion, but please feel free to open it up again if you feel like more discussion is necessary.
